pennylane/ops/qubit/hamiltonian.py

```diff
@@ -211,7 +211,7 @@ def __init__(
         super().__init__(*coeffs_flat, wires=self._wires, id=id, do_queue=do_queue)
 
     def label(self, decimals=None, base_label=None):
-        return super().label(decimals=decimals, base_label=base_label or "𝓗")
+        return super().label(decimals=None, base_label=base_label or "𝓗")
```

Comment:
For other gates, the number of parameters that might get printed is three. Especially for QChem Hamiltonians, we can end up with a lot of coefficients. For example:

```
0: ───┤ ╭<𝓗(0.10,0.20,0.30,0.40)>
1: ───┤ ╰<𝓗(0.10,0.20,0.30,0.40)>
```
And that's still a small Hamiltonian case. I'm not a big fan of how that looks.  To avoid it looking like a disaster for larger Hamiltonians, users would be restricted to drawing everything with `decimals=None`, and I don't want that.

Reply:
We could draw out the parameters when there is less than a certain number of Hamiltonian parameters, and otherwise chop it off.

Reply:
That might be nice compromise. I wonder if it would be good to handle long parameter lists like we do matrix arguments (if support for printing matrices is added).